packages/ast-spec/src/declaration/VariableDeclaration/spec.ts

```diff
@@ -63,30 +103,28 @@ export interface UsingInNormalContextDeclaration extends BaseNode {
   kind: 'await using' | 'using';
 }
 
-export interface UsingInForOfDeclaration extends BaseNode {
-  type: AST_NODE_TYPES.VariableDeclaration;
+export interface UsingInNormalContextDeclaration extends UsingDeclarationBase {
   /**
    * The variables declared by this declaration.
-   * Note that there may be 0 declarations (i.e. `const;`).
+   * Always non-empty.
    * ```
-   * for(using x of y){}
+   * using x = 1;
+   * using y = 1, z = 2;
    * ```
    */
-  // TODO - this should be guaranteed to have 1 element in it.
-  declarations: UsingInForOfDeclarator[];
-  /**
-   * This value will always be `false`
-   * because 'declare' modifier cannot appear on a 'using' declaration.
-   */
-  declare: false;
+  // TODO(#1852) - this should be guaranteed to have at least 1 element in it.
+  declarations: UsingInNomalContextDeclarator[];
```

Comment:
could we strictify this to this:
```ts
declarations: [UsingInNomalContextDeclarator, ...UsingInNomalContextDeclarator[]];
```

That should declare an array with at least 1 element and works in concert with `noUncheckedIndexedAccess`:
```ts
// both valid
const x1: [number, ...number[]] = [1];
const x2: [number, ...number[]] = [1, 2, 3, 4];

const y1 = x1[0];
//    ^? number
const y2 = x1[1];
//    ^? number | undefined
```

WDYT?

Reply:
Thought about it but:

- The utility is very low, because unless you want specifically the first declarator, it doesn't matter if the array is empty
- Tuples play badly with many type operations like inference, assignability, etc.

Reply:
All good. If we're not planning on updating the types we can delete the todo entirely then